File: AoC_2021/day_1/code.py
# star 1
with open('data.txt', 'r') as f:
	nrs = f.readlines()

nrs = [int(l) for l in nrs]
next_larger = [next for prev, next in zip(nrs[:-1], nrs[1:]) if next > prev]

print(len(next_larger))

# start 2
sums = [a + b + c for a,b,c in zip(nrs[:-2], nrs[1:-1], nrs[2:])]
next_larger_sums = [next for prev, next in zip(sums[:-1], sums[1:]) if next > prev]

print(len(next_larger_sums))

File: AoC_2021/day_1/code.R
# star 1
nrs <- readLines("~/Documents/advent_of_code/AoC_2021/day_1/data.txt") |>
  as.integer()

print(sum(nrs[-1] > nrs[-length(nrs)]))

# star 2
l <- length(nrs)
nrs_sum <- nrs[-c(1:2)] + nrs[-c(1, l)] + nrs[-c(l-1, l)]
print(sum(nrs_sum[-1] > nrs_sum[-length(nrs_sum)]))